http-server/src/main/java/io/airlift/http/server/HttpServer.java

```diff
@@ -97,8 +92,8 @@
     private final DelimitedRequestLog requestLog;
     private ConnectionStats httpConnectionStats;
     private ConnectionStats httpsConnectionStats;
-
-    private final Optional<ZonedDateTime> certificateExpiration;
+    private final ScheduledExecutorService scheduledExecutorService = newScheduledThreadPool(1);
```

Comment:
This can use `newSingleThreadExecutor()` since there's only one thread. Also, we should name the thread and use a daemon thread:
```java
newSingleThreadExecutor(daemonThreadsNamed("HttpServerScheduler"));

Reply:
We should only start this if needed -- we don't need the executor running if we're just doing HTTP.

Reply:
Updated.

Reply:
Did you forget to push? I don't see any change here.

I'm also fine merging this as-is and addressing this in a follow up.

Reply:
🤦 Yes. Fixed.